runtime/src/main/cpp/Memory.cpp

```diff
@@ -104,6 +109,10 @@ volatile int aliveMemoryStatesCount = 0;
 
 KBoolean g_checkLeaks = KonanNeedDebugInfo;
 
+// Only used by the leak detector.
+KRef g_leakCheckerGlobalList = nullptr;
+KInt g_leakCheckerGlobalLock = 0;
```

Comment:
* Global visibility is not needed here. Consider RAII + internal linkage. 
* leakCheckerGlobalList shall be accessed in the locked scope only, so make it obvious.
```
class LeakCheckerGlobalLocker {
  LeakCheckerGlobalLocker() { lock(lock_); }
  ~LeakCheckerGlobalLocker() { unlock(lock_); }
  KRef checkerGlobalList() const { return leakCheckerGlobalList_; } // still may escape, but better than nothing
private:
  LeakCheckerGlobalLocker(const LeakCheckerGlobalLocker&) = delete;
  LeakCheckerGlobalLocker(LeakCheckerGlobalLocker&&) = delete;
  static KInt lock_;
  static KRef leakCheckerGlobalList_;
};
```

Reply:
It is not global, it's anonymous C++ namespace.

Reply:
Yes, we could do RAII here, although it's not too important.